sources/constraint-definition.asciidoc

```diff
@@ -436,7 +436,7 @@ If left undefined, the default value for [methodname]`@OverridesAttribute.name`
 
 [NOTE]
 ====
-[tck-testable]#A composing constraint can itself be a composed constraint. In this case, attribute values are overridden recursively according to the described rules.# Note however, that a forwarding rule (as defined by [classname]`@OverridesAttribute`) is only applied to the direct composing constraints.
```

Comment:
Well, I think it makes sense either way. What I'm saying is that `@NotEmpty` can also be used as a composing constraint. It's usually the way you would have it: you have an existing composed constraint and you want to use it as a composing constraint.

I think it makes more sense this way but I don't have a strong opinion about it. Feel free to ignore this part if you prefer the existing version. Both are correct.